internal/authentication/ldap_user_provider.go

```diff
@@ -114,71 +120,65 @@ func (p *LDAPUserProvider) getUserAttribute(conn LDAPConnection, username string
 		baseDN = p.configuration.AdditionalUsersDN + "," + baseDN
 	}
 
+	attributes := []string{"dn",
+		p.configuration.MailAttribute,
+		p.configuration.UsernameAttribute}
```

Comment:
I was going to suggest doing multi-attribute lookups, nice!

Reply:
:)